vars/createUploadTestReportManifest.groovy

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+/** Library to close GitHub issue across opensearch-project repositories.
+ @param Map args = [:] args A map of the following parameters
+ @param args.testManifest <required> - The path of the test manifest used
+ @param args.buildManifest <required> - The path of the build manifest of OpenSearch artifacts.
+ @param args.buildManifestDashboards <optional> - The path of the build manifest of OpenSearch Dashboards
+ @param args.testRunID <required> - Test run id of the test workflow being reported.
+ @param args.testType <required> - Type of the test workflow being reported.
+ @param args.componentName <Optional> - Components that workflow runs on.
+ */
+
+def call(Map args = [:]) {
+    lib = library(identifier: 'jenkins@main', retriever: legacySCM(scm))
+
+    def testID = args.testRunID;
+    def testType = args.testType;
+
+    def testManifest = lib.jenkins.TestManifest.new(readYaml(file: args.testManifest))
+    def buildManifest = lib.jenkins.BuildManifest.new(readYaml(file: args.buildManifest))
+    def buildManifestDashboards = args.buildManifestDashboards ? lib.jenkins.BuildManifest.new(readYaml(file: args.buildManifestDashboards)) : null;
```

Comment:
nit: Does it make sense to rename to `dashboardsBuildManifest`?

Reply:
Makes sense to me. Updated the name.